src/server/game/Miscellaneous/Formulas.h

```diff
@@ -134,26 +133,33 @@ namespace Trinity
                     break;
             }
 
+            uint32 baseGainScaled;
             if (mob_level >= pl_level)
             {
                 uint8 nLevelDiff = mob_level - pl_level;
                 if (nLevelDiff > 4)
                     nLevelDiff = 4;
 
-                baseGain = ((pl_level * 5 + nBaseExp) * (20 + nLevelDiff) / 10 + 1) / 2;
+                baseGainScaled = ((pl_level * 5 + nBaseExp) * (20 + nLevelDiff) / 10 + 1) / 2;
             }
             else
             {
                 uint8 gray_level = GetGrayLevel(pl_level);
                 if (mob_level > gray_level)
                 {
                     uint8 ZD = GetZeroDifference(pl_level);
-                    baseGain = (pl_level * 5 + nBaseExp) * (ZD + mob_level - pl_level) / ZD;
+                    baseGainScaled = (pl_level * 5 + nBaseExp) * (ZD + mob_level - pl_level) / ZD;
                 }
                 else
-                    baseGain = 0;
+                    baseGainScaled = 0;
             }
 
+            // Calculate unscaled based gain min.  Use mob level instead of player level to avoid overscaling on gain
+            uint32 baseGainUnscaledMin = (mob_level * 5 + nBaseExp) * sWorld->getIntConfig(CONFIG_MIN_CREATURE_SCALED_XP_RATIO) / 100;
```

Comment:
Move this and the `std::max` call into an if branch checking your world config to make it explicit that this is custom option code.

Also, remove the rename above (leave `baseGain`, replace its value in the if branch if appropriate).